### PR TITLE
[SNOW-1320248]: Fix `inplace=True` on Series objects derived from Series.

### PR DESCRIPTION
**Pull request by sfc-gh-rdurrani**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1320248

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

This PR fixes the inplace argument for Series functions where the Series is derived from another Series object (e.g. series = series.iloc[:4]; series.fillna(14, inplace=True))

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/extensions/series_overrides.py [hunk 1]
```diff
@@ -392,6 +392,25 @@ def __init__(
...
+
+    Parameters
+    ----------
+    new_query_compiler : BaseQueryCompiler
+        QueryCompiler to use to manage the data.
+    """
+    super(Series, self)._update_inplace(new_query_compiler=new_query_compiler)
+    # Propagate changes back to parent so that column in dataframe had the same contents
+    if self._parent is not None:
+        if self._parent_axis == 1 and isinstance(self._parent, DataFrame):
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/extensions/series_overrides.py:408]: Can the parent ever be an Index? I don't think so, but just checking

Comment (sfc-gh-rdurrani) [reply src/snowflake/snowpark/modin/plugin/extensions/series_overrides.py:408]: Good question - I don't think so - I checked `to_series` and it doesn't seem to set the parent:
```python
In [1]: im();
a =
In [2]: a = pd.Index([1, 2, 3, 4])

In [3]: a.to_series()
Out[3]: a
1    1
2    2
3    3
4    4
dtype: int64

In [4]: a.to_series()._parent
```
Additionally, Index's have no `iloc`, and indexing returns another Index object, so I don't think it should be possible?
```
In [1]: im();
a =
In [2]: a = pd.Index([1, 2, 3, 4])

In [3]: type(a.iloc[:2])
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 type(a.iloc[:2])

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/extensions/index.py:282, in Index.__getattr__(self, key)
    278     if hasattr(native_index, key):
    279         # Any methods that not supported by the current Index.py but exist in a
    280         # native pandas index object should raise a not implemented error for now.
    281         ErrorMessage.not_implemented(f"Index.{key} is not yet implemented")
--> 282 raise err

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/extensions/index.py:274, in Index.__getattr__(self, key)
    256 """
    257 Return item identified by `key`.
    258
   (...)
    271 of current scope that are not implemented.
    272 """
    273 try:
--> 274     return object.__getattribute__(self, key)
    275 except AttributeError as err:
    276     if not key.startswith("_"):

AttributeError: 'Index' object has no attribute 'iloc'

In [4]: type(a[:2])
Out[4]: snowflake.snowpark.modin.plugin.extensions.index.Index
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 - Fixed a bug where an `Index` object created from a `Series`/`DataFrame` incorrectly updates the `Series`/`DataFrame`'s index name after an inplace update has been applied to the original `Series`/`DataFrame`.
 - Suppressed an unhelpful `SettingWithCopyWarning` that sometimes appeared when printing `Timedelta` columns.
-
+- Fixed `inplace` argument for Series objects derived from Series objects.
```
Comment (sfc-gh-helmeleegy) [inline CHANGELOG.md:21]:
```suggestion
- Fixed `inplace` argument for `Series` objects derived from other `Series` objects.
```

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM, thanks Rehan!

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM, please resolve other comments

**Review by sfc-gh-joshi — APPROVED** [review]
Can you check if this is a bug in OSS Modin, and file an issue there if so?

**Comment by sfc-gh-rdurrani** [comment]
> Can you check if this is a bug in OSS Modin, and file an issue there if so?

I was able to confirm that the bug is not reproducible on OSS Modin on Ray in the latest version installable from pip, so I think this is a snowpark only issue!

```python
In [1]: import modin.pandas as pd
d
In [2]: df = pd.read_csv("/Users/rdurrani/Downloads/test.csv")
2024-09-17 13:17:51,513	INFO worker.py:1749 -- Started a local Ray instance.

In [3]: s = df['Fare'].copy().iloc[:4]
   ...: s.fillna(14, inplace=True)

In [4]: s
Out[4]:
0    7.8292
1    7.0000
2    9.6875
3    8.6625
Name: Fare, dtype: float64
```

**Pull request merged**